### PR TITLE
Support blink.user_timing events in audit results (#364)

### PR DESCRIPTION
**Pull request by mgerakis**
#364

### CODE REVIEW

File: src/audits/performance/user-timings.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          args: ut.args
+        };
+      });
+    }
+
+    return UserTimings.generateAuditResult({
+      value: true,
+      extendedInfo: {
+        /* Pass empty array rather than undefined */
+        value: userTimings || []
```
Comment (paullewis) [inline src/audits/performance/user-timings.js]: We need some way of handling this in the report. Are you doing that as part of this PR, or is someone else taking it?

**Comment by paullewis** [comment]
Looks great to me, just need to know how we output the data in the reports (though for JSON we should be gtg right?)

**Comment by mgerakis** [comment]
The output isn't set up yet, and it isn't in the JSON either. I believe the plan is to have the aggragater from #321 handle this audit as well.

**Comment by paulirish** [comment]
aggregator is in!

File: src/audits/performance/user-timings.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    if (!artifacts.traceContents || !Array.isArray(artifacts.traceContents)) {
+      throw new Error(FAILURE_MESSAGE);
+    }
+
+    // Fetch blink.user_timing events from the tracing data
+    const model = new TimeLineModel(artifacts.traceContents);
+    const tModel = model.timelineModel();
```
Comment (zhaoz) [inline src/audits/performance/user-timings.js]: const timelineModel

**Comment by paulirish** [comment]
This is what it looks like when there are userTimings in the trace:

image

**Comment by paulirish** [comment]
lgtm

+1 to 

* `tModel` -> `timelineModel` and 
* `TimeLineModel` -> `TimelineModel`

File: formatters/partials/user-timings.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  .ut-measure_listing-duration {
+    font-weight: bold
+  }
+</style>
+
+<div>
+  <div class="ut-measures">
+    {{#each this}}
+      <div>
+        {{ this.name }} - <span class="ut-measure_listing-duration">{{ this.duration }}</span>
```
Comment (paulirish) [inline formatters/partials/user-timings.html]: I'd swap these.  `duration - name`  as it'll make it easier to compare

**Pull request merged**